src/header/NamespaceDropdown/NamespaceDropdown.tsx

```diff
@@ -1,18 +1,49 @@
 import { Menu, Icon } from 'fundamental-react';
-import { Link } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+import { Link, useLocation } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 
+import { namespacesState } from 'state/namespacesAtom';
 import { useUrl } from 'hooks/useUrl';
-import { NamespacesState } from 'state/namespacesAtom';
 
 import './NamespaceDropdown.scss';
 
+const Namespaces = () => {
+  const namespaces = useRecoilValue(namespacesState);
+  const { clusterUrl, namespace, namespaceUrl } = useUrl();
+  const { pathname } = useLocation();
+
+  const switchNamespace = (namespaceName: string) => {
+    if (!pathname.includes('/namespaces/'))
+      return clusterUrl(`namespaces/${namespaceName}`);
+
+    const pathElements = pathname
+      .substring(pathname.indexOf('namespaces/'))
+      .split('/');
+
+    const resourceType = pathElements[2];
+    const resourceName = pathElements[3];
```

Comment:
i don't like this manual splitting - i think it would be better to use `useMatch` - similar to how it's used inside `useUrl`. Something similar to this:
```suggestion
    const {resourceType, resourceName } = useMatch({ path: '/cluster/:cluster/namespaces/:namespace/:resourceType/:resourceName', end: false })?.params
```